File: lab_12/calculation.py
def main():
    a = float(input("Enter first number: "))
    b = float(input("Enter second number: "))

    sum_result = a + b
    diff_result = a - b
    prod_result = a * b
    quot_result = a / b

    with open("calculation.txt", "w") as file:
        file.write(f"{a} + {b} = {sum_result}\n")
        file.write(f"{a} - {b} = {diff_result}\n")
        file.write(f"{a} * {b} = {prod_result}\n")
        file.write(f"{a} / {b} = {quot_result}\n")

    print("[บันทึกผลลัพธ์เรียบร้อย!]")

    print("\n[อ่านข้อมูลจากไฟล์ calculation.txt]")
    with open("calculation.txt", "r") as file:
        for line in file:
            print(line.strip())

if __name__ == "__main__":
    main()

File: lab_12/calculation.cpp
#include <iostream>
#include <fstream>
using namespace std;

int main() {
    double a, b;
    cout << "Enter first number: ";
    cin >> a;
    cout << "Enter second number: ";
    cin >> b;

    double sum = a + b;
    double diff = a - b;
    double prod = a * b;
    double quot = a / b;

    ofstream outfile("calculation.txt");
    outfile << a << " + " << b << " = " << sum << endl;
    outfile << a << " - " << b << " = " << diff << endl;
    outfile << a << " * " << b << " = " << prod << endl;
    outfile << a << " / " << b << " = " << quot << endl;
    outfile.close();

    cout << "[บันทึกผลลัพธ์เรียบร้อย!]" << endl;

    ifstream infile("calculation.txt");
    string line;
    cout << "\n[อ่านข้อมูลจากไฟล์ calculation.txt]" << endl;
    while (getline(infile, line)) {
        cout << line << endl;
    }
    infile.close();

    return 0;
}
